searx/weather.py

```diff
@@ -0,0 +1,610 @@
+            value=value,
+        )
+
+    def __str__(self):
+        return self.l10n()
+
+    def value(self, unit: Units) -> float:
+        return convert_from_si(si_name=self.si_name, symbol=unit, value=self.si)
+
+    def l10n(
+        self,
+        unit: Units | None = None,
+        locale: babel.Locale | GeoLocation | None = None,
+        template: str = "{value} {unit}",
+        num_pattern: str = "#,##0",
+    ) -> str:
+        """Localized representation of a measured value.
+
+        If the ``unit`` is not set, an attempt is made to determine a ``unit``
+        matching the territory of the ``locale``.  If the locale is not set, an
+        attempt is made to determine it from the HTTP request.
+
+        The value is converted into the respective unit before formatting.
+
+        The argument ``num_pattern`` is used to determine the string formatting
+        of the numerical value:
+
+        - https://babel.pocoo.org/en/latest/numbers.html#pattern-syntax
+        - https://unicode.org/reports/tr35/tr35-numbers.html#Number_Format_Patterns
+
+        The argument ``template`` specifies how the **string formatted** value
+        and unit are to be arranged.
+
+        - `Format Specification Mini-Language
+          <https://docs.python.org/3/library/string.html#format-specification-mini-language>`.
+        """
+
+        if isinstance(locale, GeoLocation):
+            locale = locale.locale()
+        elif locale is None:
+            locale = babel.Locale.parse(_get_sxng_locale_tag(), sep='-')
+
+        if unit is None:  # unit by territory
+            unit = "°C"
+            if locale.territory in ["US"]:
+                unit = "°F"
+        val_str = babel.numbers.format_decimal(self.value(unit), locale=locale, format=num_pattern)
+        return template.format(value=val_str, unit=unit)
+
+
+class Pressure:
+    """Class for converting pressure units and for string representation of
+    measured values."""
+
+    si_name = "Q44395"
+
+    Units = typing.Literal["Pa", "hPa", "cm Hg", "bar"]
+    """Supported units."""
+
+    units = list(typing.get_args(Units))
+
+    def __init__(self, value: float, unit: Units):
+        if unit not in self.units:
+            raise ValueError(f"invalid unit: {unit}")
+        # pylint: disable=invalid-name
+        self.si: float = convert_to_si(si_name=self.si_name, symbol=unit, value=value)
+
+    def __str__(self):
+        return self.l10n()
+
+    def value(self, unit: Units) -> float:
+        return convert_from_si(si_name=self.si_name, symbol=unit, value=self.si)
+
+    def l10n(
+        self,
+        unit: Units | None = None,
+        locale: babel.Locale | GeoLocation | None = None,
+        template: str = "{value} {unit}",
+        num_pattern: str = "#,##0",
+    ) -> str:
+        if isinstance(locale, GeoLocation):
+            locale = locale.locale()
+        elif locale is None:
+            locale = babel.Locale.parse(_get_sxng_locale_tag(), sep='-')
```

Comment:
This could probably be moved to a helper function (or `_get_sxng_locale_tag()`)? It's the same in all `l10n` methods.

Reply:
In the currently implemented six classes, the default locale is determined from the query. However, we may eventually want to implement classes that determine their default locale solely from the UI property (where the language of the query is irrelevant). We can implement an additional helper function, as you suggested, but that would only be a wrapper that would then have to call [_get_sxng_locale_tag](https://github.com/searxng/searxng/pull/4663#discussion_r2106152038). That would be a few fewer lines of code, but I wasn't sure if it was worth it. After all, it would be another function.

If you find an additional helper function helpful, we can implement it.

Reply:
> If you find an additional helper function helpful, we can implement it.
>
You're right, it probably doesn't make too much sense looking at the code again. I've only seen that there's duplicated code and didn't really look deeper into why it's duplicated here :)